subsystem/ams_subscriptions_test.go

```diff
@@ -0,0 +1,193 @@
+package subsystem
+
+import (
+	"context"
+	"net/http"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/assisted-service/client/installer"
+	"github.com/openshift/assisted-service/models"
+)
+
+var _ = Describe("test AMS subscriptions", func() {
+	ctx := context.Background()
+
+	BeforeEach(func() {
+		if !Options.EnableAuth {
+			Skip("auth is disabled")
+		}
+		if !Options.WithAMSSubscriptions {
+			Skip("AMS is disabled")
+		}
+	})
+
+	AfterEach(func() {
+		clearDB()
+	})
+
+	Context("AMS subscription on cluster creation", func() {
+
+		It("happy flow", func() {
+
+			// create subscription
+			clusterID, err := registerCluster(ctx, userBMClient, "test-ams-subscriptions-cluster", pullSecret)
+			Expect(err).ToNot(HaveOccurred())
+			log.Infof("Register cluster %s", clusterID)
+			cc := getCommonCluster(ctx, clusterID)
+			Expect(cc.AmsSubscriptionID).To(Equal(FakeSubscriptionID))
+
+			// update subscription with openshfit (external) cluster ID
+			registerHostsAndSetRoles(clusterID, 3)
+			c := installCluster(clusterID)
+			for _, h := range c.Hosts {
+				updateProgress(*h.ID, clusterID, models.HostStageDone)
+			}
+			waitForClusterState(ctx, clusterID, models.ClusterStatusFinalizing, defaultWaitForClusterStateTimeout, clusterFinalizingStateInfo)
+			err = completeInstallation(agentBMClient, clusterID)
+			Expect(err).ToNot(HaveOccurred())
+			waitForClusterState(ctx, clusterID, models.ClusterStatusInstalled, defaultWaitForClusterStateTimeout, "installed")
+		})
+
+		// ATTENTION: this test override a wiremock stub - DO NOT RUN IN PARALLEL
```

Comment:
wonder if there's some fixture to avoid running it in parallel maybe :/

Reply:
This would have been a great idea!
Didn't find anything on google though.